File: 1897. Redistribute Characters to Make All Strings Equal.java
class Solution {
    public boolean makeEqual(String[] words) {
        int p = words.length;
        int[] count = new int[26];
        for (String word : words) {
            for (char ch : word.toCharArray()) {
                count[ch - 'a']++;
            }
        }
        for (int k = 0; k < 26; k++) {
            if (count[k] % p != 0) {
                return false;
            }
        }
        return true;
    }
}

File: 1897. Redistribute Characters to Make All Strings Equal.cpp
class Solution {
public:
    bool makeEqual(vector<string>& words) {
        vector<int> freq(26); 
        for (auto& word : words) 
            for (auto& ch : word) ++freq[ch-'a']; 
        for (int i = 0; i < 26; ++i) 
            if (freq[i] % size(words)) return false; 
        return true; 
    }
};
